### DISCUSSION TITLE
Tuya - Neos Gateway Hub(Zigbee)

### DISCUSSION THREAD

**Discussion by isarib01** [make-all/tuya-local#972] [General]
Hi,

I purchased a Neo Gateway Hub for Zigbee devices(usb powered), which connects to my network over ethernet. I have a bunch of Temp/Humidity(battery) and Door Read Sensors(battery).  These are all linked up and paired into my Tuya Device App nicely.  I am not able to see any of these devices in my Tuya IoT Platform project...  I can see all my other Tuya compatible devices fine..

Is there a way for Tuya-local to get around the IoT platform and talk directly to the Zigbee devices or even the Neo Gateway Hub? or am I not doing something correct?

FYI - For all my other Tuya devices, they work fine with the IoT platform, and Tuya-local.

**Comment by make-all** [comment]
You will need some way to get the local key from the devices, many of the easier options require iot platform access.  You say "Tuya Device App" - do you mean the generic Tuya Smart App, or the manufacturer's branded app?  It may just be a case of needing to pair your devices with the generic app.

**Comment by isarib01** [comment]
Hi,
"Tuya Device App" is in reference to my smartphone "Tuya Smart App".
Regards 
____________________________________________ 
Imran
    On Tuesday, 8 August 2023 at 08:19:26 pm AWST, Jason Rumney ***@***.***> wrote:  
 
 


You will need some way to get the local key from the devices, many of the easier options require iot platform access. You say "Tuya Device App" - do you mean the generic Tuya Smart App, or the manufacturer's branded app? It may just be a case of needing to pair your devices with the generic app.

**Comment by isarib01** [comment]
Hi,

I have managed to get the device id, and local key for the Neo Gateway Hub, which I partly got from the python tinytuya scan, and entered the device id into the Tuya IoT developers platform, and query the device.  Even though Tuya IoT does not show the device in the list, but if you entered the device id into the query device section, you can get the local keys etc...

Now I am stuck at using tinytuya wizard to extract the Sub Device/Node ID's for all the zigbee devices...  Is this the only method to extract sub device id's?

![image](https://github.com/make-all/tuya-local/assets/73514261/c89abaaf-8c6b-448d-b8be-fd7a830488e6)

**Comment by isarib01** [comment]
Managed to sort it out!, seems like in Tuya IoT, I was able to add all the Gateway and Sub Devices by "linking" on all the items of the zigbee's.  Once that was done, I could see all my devices and also run a query.

1) I have an issue out of all the zigbee devices, the door read/magnetic sensor does not appear to be supported in the Tuya Local, the only options are below:

![image](https://github.com/make-all/tuya-local/assets/73514261/7e90c72d-4771-45d4-b6d9-eb9053cf138d)

2) The Temp/Humidity sensor displays the humidty in the 100's%, ie 650%.

3) I seem to think there is a limit of 3 Sub Devices I can add.  I hit an error whilst trying to add anymore..

![image](https://github.com/make-all/tuya-local/assets/73514261/e409c775-3ac6-4582-95a0-062029b43e3e)

**Comment by make-all** [comment]
1. There are some door/window sensors in the queue that might solve this, but to make sure you could file a new device request yourself with the log info (or check the matching issues yourself)
2. The just added ZBSC10WT sensor may resolve this.  It has dp 1=temperature, dp 2=humidity, dp 4=battery % all scaled by 10. (just added means AFTER 2023.8.1 was released, so not yet in a release).
3. This is a limit of the number of connections your hub supports.  Currently this integration is not smart enough to share a connection between all the devices connected to the same hub.

**Reply by isarib01** [reply]
1) and 2), will try to see what you currently have available, as well as wait for the 2023.8.1
3). Are you aware of any work arounds? or firmware which could override these limitations?  Are other Zigbee gateways behaving like this? - if not, what do you recommend?

**Reply by make-all** [reply]
2023.8.1 was released a couple of days after my last comment.  2023.8.2 is the current version now.

There is no workaround for the hub connection limitation, other than ditching the Tuya zigbee hub and using another zigbee hub, or USB stick with ZHA or Zigbee2MQTT integrations instead of this one, or using the Tuya cloud integration if that works for the devices.

**Reply by isarib01** [reply]
Regarding 2) 

I have managed to update to 2003.9.1, and noticed the Temp is now displaying the correct degree's.  However the Humidty is still indicating in the 100's, ie 600% :)
